presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -739,6 +739,48 @@ private void revokeRole(String role, String granteeName, PrincipalType granteeTy
         }
     }
 
+    private List<String> getPrestoViews(String databaseName)
+            throws TException
+    {
+        /*
+         * Thrift call `get_table_names_by_filter` may be translated by Metastore to a SQL query against Metastore database.
+         * Hive 2.3 on some databases uses CLOB for table parameter value column and some databases disallow `=` predicate over
+         * CLOB values. At the same time, they allow `LIKE` predicates over them.
+         */
+        String filterWithEquals = HIVE_FILTER_FIELD_PARAMS + PRESTO_VIEW_FLAG + " = \"true\"";
+        String filterWithLike = HIVE_FILTER_FIELD_PARAMS + PRESTO_VIEW_FLAG + " LIKE \"true\"";
+
+        if (metastoreKnownToSupportTableParamEqualsPredicate) {
+            try (ThriftMetastoreClient client = clientProvider.createMetastoreClient()) {
+                return client.getTableNamesByFilter(databaseName, filterWithEquals);
+            }
+        }
+        if (metastoreKnownToSupportTableParamLikePredicate) {
+            try (ThriftMetastoreClient client = clientProvider.createMetastoreClient()) {
+                return client.getTableNamesByFilter(databaseName, filterWithLike);
+            }
+        }
+
+        try (ThriftMetastoreClient client = clientProvider.createMetastoreClient()) {
+            List<String> views = client.getTableNamesByFilter(databaseName, filterWithEquals);
+            metastoreKnownToSupportTableParamEqualsPredicate = true;
+            return views;
+        }
+        catch (TException | RuntimeException firstException) {
+            try (ThriftMetastoreClient client = clientProvider.createMetastoreClient()) {
+                List<String> views = client.getTableNamesByFilter(databaseName, filterWithLike);
+                metastoreKnownToSupportTableParamLikePredicate = true;
+                return views;
+            }
+            catch (TException | RuntimeException secondException) {
+                if (firstException != secondException) {
```

Comment:
Technically you can keep an exception and rethrow it multiple times. I see no particular reason why this could happen here.
Apparently we often guard calls to `java.lang.Throwable#addSuppressed` like this (for apparently no particular reason), so I am blindly following an established convention 😉

Reply:
It seems unlikely to happen in practice but is a good safety measure.